### PR TITLE
add code samples and compatibility notes

### PR DESCRIPTION
**Pull request by galkoren**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
 
-```
-Only MySQL is currently supported.
-```
 
-## Usage
+## Compatibility
+* Java 8 or greater
+* MySQL Database
+* Each PL release version specifis the JOOQ version it was built with. E.g. release "0.1.40-jooq-3.10.4" was built with JOOQ 3.10.4. If you need a very specific JOOQ version, we can add it to our build process.
```
Comment (effiban) [inline README.md:19]: typo: specifis -> specifi**e**s

File: README.md [hunk 2]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
+
+## Code Samples
 
+### Retrieve auto generated ID
 After defining and wirnig the required infra (as described in the Book below), you can execute PL commands as follows:
```
Comment (effiban) [inline README.md:24]: typo: wirnig -> wir**in**g

File: README.md [hunk 3]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
 ```java
-var cmd = new CreateCampaignCmd(someId);
+var cmd = new CreateCampaignCommand();
```
Comment (effiban) [inline README.md:26]: I see you're supporting Java 11 already in the examples :) Was this intended?

File: README.md [hunk 4]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
+
+campaignCmd.addChild(new UpsertCityCommand("London"));
+campaignCmd.addChild(new UpsertCityCommand("Tel Aviv"));
+campaignCmd.addChild(new UpsertCityCommand("Miami"));
+campaignCmd.add(new DeletionOfOther(CityEntity.INSTANCE));
+
+campaignPersistence.update(asList(campaignCmd));
 ```
+* The campaign ID is automatically populated into the campaign_cities table.
+* All other cities of this campaign are deleted from the DB.
```
Comment (effiban) [inline README.md:51]: :+1:

File: README.md [hunk 5]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
+    public static final EntityField<Campaign, Integer> BIDDING_STRATEGY = INSTANCE.field(CampaignBiddingStrategy.TABLE.strategy_type);
+}
+```
+
+There are multiple features in this examples:
+* ID field is auto incremented by the database (by JOOQ expression ```identity.true()```).
+* ID field cannot be changed by an UPDATE command.
+* BUDGET field must be provided upon creation.
+* ACCOUNT_ID must be provided upon creation and the referenced Account entity must exist in DB.
+* BIDDING_STRATEGY is a field from another JOOQ table (usages of the campaign entity should not care about it). This requires that the "secondary" table referring the campaing table by a one-to-one relation.
```
Comment (effiban) [inline README.md:106]: typo/grammar: "referring the campaing table" -> "ref**er to** the campa**ig**n table"

File: README.md (hunk 3 above)
Comment (galkoren) [reply README.md:26]: For the sake of the example (which should be short), my imaginary user prefers Java 11.

Comment (effiban) [reply README.md:26]: sounds good to me...

File: README.md [hunk 6]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
+## Compatibility
+* Java 8 or greater
+* MySQL Database
+* Each PL release version specifies the JOOQ version it was built with. E.g. release "0.1.40-jooq-3.10.4" was built with JOOQ 3.10.4. If you need a very specific JOOQ version, we can add it to our build process.
+
+## Code Samples
 
-After defining and wirnig the required infra (as described in the Book below), you can execute PL commands as follows:
+### Retrieve auto generated ID
+After defining and wirinig the required infra (as described in the Book below), you can execute PL commands as follows:
```
Comment (effiban) [inline README.md:24]: typo still exists: wirinig ->  **wiring**

File: README.md [hunk 7]
```diff
@@ -12,22 +12,98 @@ PL is a Java mutation layer for business entites where you can define a flow to
...
+    public static final EntityField<Campaign, Integer> BIDDING_STRATEGY = INSTANCE.field(CampaignBiddingStrategy.TABLE.strategy_type);
+}
+```
+
+There are multiple features in this examples:
+* ID field is auto incremented by the database (by JOOQ expression ```identity.true()```).
+* ID field cannot be changed by an UPDATE command.
+* BUDGET field must be provided upon creation.
+* ACCOUNT_ID must be provided upon creation and the referenced Account entity must exist in DB.
+* BIDDING_STRATEGY is a field from another JOOQ table (usages of the campaign entity should not care about it). This requires that the "secondary" table refer to the campaing table by a one-to-one relation.
```
Comment (effiban) [inline README.md:106]: typo near the end still exists: campaing -> campai**gn**

File: README.md (hunk 6 above)
Comment (galkoren) [reply README.md:24]: damn it!

File: README.md (hunk 7 above)
Comment (galkoren) [reply README.md:106]: how can you notice these things?!

**Pull request merged**